### PR TITLE
Allows Settings to be defined in YAML

### PR DESCRIPTION
**Pull request by firemankurt**
Allow settings array to be defined in the plugin.yaml file instead of just in php file.

### CODE REVIEW

**Comment by LukeTowers** [comment]
Don't forget to only make PRs to `develop`. I'll change it for you this time but keep it in mind for the future.

File: modules/system/classes/PluginBase.php [hunk 1]
```diff
@@ -138,7 +138,10 @@ public function registerPermissions()
      */
     public function registerSettings()
     {
-        return [];
+		$configuration = $this->getConfigurationFromYaml();
```
Comment (LukeTowers) [inline modules/system/classes/PluginBase.php]: Project standard is spaces, not tabs. Please fix and then we can merge this.

**Comment by firemankurt** [comment]
Converted all indents to spaces as requested

**Comment by LukeTowers** [comment]
Perfect, thanks!

Now that settings can be initialized in the plugin.yaml, feel up to the task of extending the RainLab.Builder plugin to allow for managing the settings of a plugin through that?

**Comment by firemankurt** [comment]
I was afraid you might ask that.   I will take a look at it when I get
time.  Thanks for your hard work.

**Comment by LukeTowers** [comment]
Gotta keep moving forward man :) Thanks for your work too!

**Pull request merged**